tests/rptest/tests/enterprise_features_license_test.py

```diff
@@ -216,14 +216,12 @@ def test_enable_features(self, feature, install_license, disable_trial):
 
         has_license = not disable_trial or install_license
 
-        # RBAC isn't controlled by cluster config and so is not subject to
-        # sanction/restriction pending CORE-8029
-        expect_rejected = not has_license and not (feature == Feature.rbac)
+        expect_rejected = not has_license
 
         if expect_rejected:
             with expect_exception(
-                    requests.exceptions.HTTPError,
-                    lambda e: FEATURE_DEPENDENT_CONFIG[
+                    requests.exceptions.HTTPError, \
+                    lambda e: e.response.status_code == 403 or FEATURE_DEPENDENT_CONFIG[
```

Comment:
nit/question: is the status code of a rejected config change non-403? Otherwise the `or FEATURE_DEPENDENT_CONFIG[...]` part is redundant.

Reply:
>is the status code of a rejected config change non-403

Yes. The response for a config patch has an errors field, so the rejection shows up as a `400 Bad Request` error (i.e. like a validation error). 

It's a good point though. I'll look at rejiggering to conditionally report 403. 

Playing devil's advocate: what should we do if a config patch contains _both_ invalid and forbidden changes?

Reply:
I was only thinking about it in terms of the validity of the test, to be honest. I think leaving the status code as is is fine.